contracts/contracts/RecoveryWithDelayPlugin.sol

```diff
@@ -0,0 +1,281 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+import {ISafe} from "@safe-global/safe-core-protocol/contracts/interfaces/Accounts.sol";
+import {ISafeProtocolPlugin} from "@safe-global/safe-core-protocol/contracts/interfaces/Integrations.sol";
+import {ISafeProtocolManager} from "@safe-global/safe-core-protocol/contracts/interfaces/Manager.sol";
+import {BasePluginWithEventMetadata, PluginMetadata} from "./Base.sol";
+import {SafeTransaction, SafeRootAccess, SafeProtocolAction} from "@safe-global/safe-core-protocol/contracts/DataTypes.sol";
+
+/**
+ * @title RecoveryWithDelayPlugin - A contract compatible with Safe{Core} Protocol that replaces a specified owner for a Safe by a non-owner account.
+ * @notice This contract should be listed in a Registry and enabled as a Plugin for an account through a Manager to be able to intiate recovery mechanism.
+ * @dev The recovery process is initiated by a recoverer account. The recoverer account is set during the contract deployment in the constructor and cannot be updated.
+ *      The recoverer account can initiate the recovery process by calling the createAnnouncement function and later when the delay is over, any account can execute
+ *      complete the recovery process by calling the executeFromPlugin function.
+ * @author Akshay Patel - @akshay-ap
+ */
+contract RecoveryWithDelayPlugin is BasePluginWithEventMetadata {
+    // Constants
+    bytes32 public constant DOMAIN_SEPARATOR_TYPEHASH = keccak256("EIP712Domain(uint256 chainId,address verifyingContract)");
+
+    bytes32 public constant DELAYED_RECOVERY_TRANSACTION_TYPEHASH =
+        keccak256(
+            "DelayedRecoveryTransaction(address recoverer,address manager,address account,address prevOwner,address oldOwner,address newOwner,uint256 nonce)"
+        );
+
+    struct Announcement {
+        uint64 executionTime; // Block time in seconds when the announced transaction can be executed
+        uint64 validityDuration; // Duration in seconds the announcement is valid after delay is over (0 is valid forever)
+        bool executed; // Flag if the announced transaction was executed
+    }
+
+    // Contract storage
+
+    // Only recoverer can initiate recovery process
+    address public immutable recoverer;
```

Comment:
nitpick: Immutables aren't technically storage, are they?

Reply:
yes. should move the comment below.
Initially had thought of keeping `recoverer` updatable.